Content.Client/Access/UI/AgentIDCardWindow.xaml.cs

```diff
@@ -1,24 +1,96 @@
+using Content.Client.Stylesheets;
+using Content.Shared.StatusIcon;
 using Robust.Client.AutoGenerated;
+using Robust.Client.GameObjects;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
 using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+using System.Linq;
+using System.Numerics;
 
 namespace Content.Client.Access.UI
 {
     [GenerateTypedNameReferences]
     public sealed partial class AgentIDCardWindow : DefaultWindow
     {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IEntitySystemManager _entitySystem = default!;
+        private readonly SpriteSystem _spriteSystem = default!;
+
+        private const string JobIconPrarentId = "JobIcon";
+        private const int JobIconColumnCount = 10;
+
         public event Action<string>? OnNameChanged;
         public event Action<string>? OnJobChanged;
+        public readonly List<Button> JobIconButtons = new();
+        public readonly Dictionary<int, string> StateByIndex = new();
 
         public AgentIDCardWindow()
         {
             RobustXamlLoader.Load(this);
+            IoCManager.InjectDependencies(this);
+            _spriteSystem = _entitySystem.GetEntitySystem<SpriteSystem>();
 
             NameLineEdit.OnTextEntered += e => OnNameChanged?.Invoke(e.Text);
             NameLineEdit.OnFocusExit += e => OnNameChanged?.Invoke(e.Text);
 
             JobLineEdit.OnTextEntered += e => OnJobChanged?.Invoke(e.Text);
             JobLineEdit.OnFocusExit += e => OnJobChanged?.Invoke(e.Text);
+
+            var jobIconGroup = new ButtonGroup();
+            var allJobIcons = _prototypeManager.EnumeratePrototypes<StatusIconPrototype>();
+            var i = 0;
+            foreach (var jobIcon in allJobIcons)
+            {
+                if (!(jobIcon.Parents?.Contains(JobIconPrarentId) ?? false))
+                {
+                    continue;
+                }
+
+                var specifier = jobIcon.Icon as SpriteSpecifier.Rsi;
+                if (specifier == null)
+                {
+                    continue;
+                }
+
+                String styleBase = StyleBase.ButtonOpenBoth;
+                var modulo = i % JobIconColumnCount;
+                if (modulo == 0)
+                    styleBase = StyleBase.ButtonOpenRight;
+                else if (modulo == JobIconColumnCount - 1)
+                    styleBase = StyleBase.ButtonOpenLeft;
+
+                // Generate buttons
+                var jobIconButton = new Button
+                {
+                    Access = AccessLevel.Public,
+                    StyleClasses = { styleBase },
+                    MaxSize = new Vector2(42, 28),
+                    Group = jobIconGroup,
+                    Pressed = i == 0,
+                    ToolTip = specifier.RsiState,
```

Comment:
```suggestion
```
RSI states aren't localized strings and shouldn't be shown to users.
Either just don't have a tooltip or IconPrototype needs to get a name data-field.